### PR TITLE
Integrate clib-package into the main repository

### LINKED ISSUE #145: pull in clib-* repos
there's no need to keep them (clib-package, clib-cache, ...) separate (they're not useful for anything other than clib itself). having multiple repos here just adds complexity when we go to release new versions/etc

### PR DESCRIPTION
**Pull request by Isty001**
This is the first iteration to resolve #145 to make future development much easier. 

I moved the sources, tests and the dependencies from clib-package into the main repository. It seems to be working for me like before, but it'd be great if someone could give this a little bit of testing.

This PR is quite large because a lot of files had to be moved, but no changes has been made besides the `Makefile` and `test.sh`.

### CODE REVIEW

**Comment by jwerle** [comment]
@Isty001 testing now!

File: test.sh [hunk 1]
```diff
@@ -28,6 +28,4 @@ fi
 cd ../../
 
 
-print $EXIT_CODE
```
Comment (jwerle) [inline test.sh]: nice ;)

Comment (Isty001) [reply test.sh]: A PR is not really a PR without debug code left in it. :D

**Comment by Isty001** [comment]
> looks like some binaries were added here

Sorry, removed them & added to the .gitignore

**Comment by jwerle** [comment]
looks good!

**Comment by Isty001** [comment]
> looks good!

Then I'm merging this one. I tested a bit more, and didn't catch any issues so far.

**Pull request merged**